ansible/roles/maintenance/tasks/main.yml

```diff
@@ -0,0 +1,108 @@
+- name: Checkout Caracalla
+  git:
+    repo: "{{caracalla_git_repo}}"
+    dest: "{{caracalla_checkout}}"
+    force: yes
+    version: "{{caracalla_branch}}"
+
+# We don't use {{caracalla_checkout}}, because we want to use master
+# branch as base branch. {{caracalla_checkout}} can use different branch
+# e.g. developer adds new functionality in feature branch like you
+# right now [jiri] :-P
+- name: Delete old caracalla git directory (PR)
+  shell: 'rm -rf {{git_dir}}'
+
+- name: Create new directory for Caracalla clone
+  shell: 'mkdir {{git_dir}}'
+
+- name: Checkout Caracalla branch (PR)
+  shell: 'git clone {{git_repo}}'
+  args:
+    chdir: "{{git_dir}}"
+
+- set_fact:
+    git_dir: "{{git_dir}}/caracalla"
+
+- name: Get current date
+  shell: 'date +%Y-%m-%d_%H-%M'
+  register: date_output
+
+- set_fact:
+    git_branch: "{{date_output.stdout}}"
+
+- set_fact:
+    current_date: "{{date_output.stdout}}"
+
+- name: Create new git branch and do checkout of this branch
```

Comment:
With this many commands/shell all one after the other, it might be better to aggregate some of this in a script, and use the ansible script module (see http://docs.ansible.com/ansible/latest/script_module.html).

I also find it easier to debug/modify a shell script.

Use your judgment to decide how much should go in the shell script. (All shell/commands after the checkouts would be appropriate in my opinion).

Ansible will use role paths for scripts, so if you put a script in `ansible/roles/maintenance/files/create_baseline_pr.sh`, then you can just refer to it here as `create_baseline_pr.sh`.

Reply:
I don't agree with this request. Let me explain my concerns. When some step failed in Ansible playbook, then rest of steps in Ansible playbook are not executed. It is good functionality in this case. Implementing similar functionality using bash would require something like this:

```bash
first_command

if [ $? -eq 0 ]
then
    second_command
else
    exit 1
fi

if [ $? -eq 0 ]
then
    third_command
else
    exit 1
fi

exit 0
```

Thoughts?